validator/server_api/json.go

```diff
@@ -146,20 +146,32 @@ func ValidationInputFromJson(entry *InputJSON) (*validator.ValidationInput, erro
 		}
 		valInput.BatchInfo = append(valInput.BatchInfo, decInfo)
 	}
-	for moduleHash, info := range entry.UserWasms {
-		asm, err := base64.StdEncoding.DecodeString(info.Asm)
-		if err != nil {
-			return nil, err
-		}
-		module, err := base64.StdEncoding.DecodeString(info.Module)
-		if err != nil {
-			return nil, err
-		}
-		decInfo := state.ActivatedWasm{
-			Asm:    asm,
-			Module: module,
+	for arch, wasms := range entry.UserWasms {
+		archWasms := make(map[common.Hash][]byte)
+		for moduleHash, encoded := range wasms {
+			decoded, err := base64.StdEncoding.DecodeString(encoded)
+			if err != nil {
+				return nil, err
+			}
+			maxSize := 2_000_000
+			var uncompressed []byte
+			for {
+				uncompressed, err = arbcompress.Decompress(decoded, maxSize)
```

Comment:
Perhaps we should use a go brotli library for this instead, since it's not in consensus and would have a nicer API? Or alternatively we could add a new decompression stream API to this.

Reply:
forgot to address this:
I think we will lose more from having multiple implementations of brotli than we'll gain from a better API in this specific case.
Also, I hope API will improve after wasmer answer and provide a reasonable upper-bound.